### PR TITLE
Fixes for poping up SolidWorks windows

### PR DESCRIPTION
**Pull request by thopiekar**

### CODE REVIEW

File: CommonComReader.py [hunk 1]
```diff
@@ -87,13 +87,13 @@ def _reader_for_file_format(self):
         
         return _reader_for_file_format
 
-    def startApp(self, visible = False):
+    def startApp(self, **options):
```
Comment (LipuFei) [inline CommonComReader.py]: There is no need for `**`, it can just be `options` which is a dictionary.

File: CommonComReader.py [hunk 2]
```diff
@@ -137,7 +137,7 @@ def _read(self, file_path):
         # Starting app and Coinit before
         comtypes.CoInitializeEx(comtypes.COINIT_MULTITHREADED)
         try:
-            options["app_instance"] = self.startApp()
+            options = self.startApp(**options)
```
Comment (LipuFei) [inline CommonComReader.py]: Same here, no need for `**`

File: SolidWorksReader.py [hunk 3]
```diff
@@ -86,18 +86,28 @@ def setAppVisible(self, state, **options):
     def getAppVisible(self, state, **options):
         return options["app_instance"].Visible
 
-    def startApp(self, visible=False):
-        app_instance = super().startApp(visible = visible)
+    def startApp(self, **options):
```
Comment (LipuFei) [inline SolidWorksReader.py]: Same here

File: SolidWorksReader.py [hunk 4]
```diff
@@ -86,18 +86,28 @@ def setAppVisible(self, state, **options):
...
+        options = super().startApp(**options)
```
Comment (LipuFei) [inline SolidWorksReader.py]: Same here

**Comment by LipuFei** [comment]
I have tested this with SolidWorks 2016 and 2017. Both seem to work fine. I will merge this.

**Comment by thopiekar** [comment]
:+1:

**Pull request merged**